### PR TITLE
[cli] [mysqlctl] Migrate mysqlctl flags to pflags

### PR DESCRIPTION
**Pull request by rsajwani**
Signed-off-by: Rameez Sajwani <rameezwazirali@hotmail.com>

## Description
Updates mysqlctl related flag references specified in https://github.com/vitessio/vitess/issues/10858

go list -f '{{ .ImportPath }}| {{ join .Deps " " }} ' ./go/cmd/... | awk -F"|" '$2 ~ "vitess.io/vitess/go/vt/mysqlct" {print $1}'
vitess.io/vitess/go/cmd/mysqlctl
vitess.io/vitess/go/cmd/mysqlctld
vitess.io/vitess/go/cmd/rulesctl
vitess.io/vitess/go/cmd/rulesctl/cmd
vitess.io/vitess/go/cmd/rulesctl/common
vitess.io/vitess/go/cmd/vtadmin
vitess.io/vitess/go/cmd/vtbackup
vitess.io/vitess/go/cmd/vtcombo
vitess.io/vitess/go/cmd/vtctl
vitess.io/vitess/go/cmd/vtctld
vitess.io/vitess/go/cmd/vtctldclient
vitess.io/vitess/go/cmd/vtctldclient/command
vitess.io/vitess/go/cmd/vtctldclient/docgen
vitess.io/vitess/go/cmd/vtexplain
vitess.io/vitess/go/cmd/vtgate
vitess.io/vitess/go/cmd/vtgateclienttest
vitess.io/vitess/go/cmd/vtgr
vitess.io/vitess/go/cmd/vtorc
vitess.io/vitess/go/cmd/vttablet
vitess.io/vitess/go/cmd/vttestserver
vitess.io/vitess/go/cmd/zk

## Related Issue(s)
#10810 

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/mysqlctl/backup.go [hunk 1]
```diff
@@ -74,25 +77,38 @@ var (
...
 	// once before the writer blocks
-	backupCompressBlocks = flag.Int("backup_storage_number_blocks", 2, "if backup_storage_compress is true, backup_storage_number_blocks sets the number of blocks that can be processed, at once, before the writer blocks, during compression (default is 2). It should be equal to the number of CPUs available for compression")
+	backupCompressBlocks = 2
 
 	backupDuration  = stats.NewGauge("backup_duration_seconds", "How long it took to complete the last backup operation (in seconds)")
 	restoreDuration = stats.NewGauge("restore_duration_seconds", "How long it took to complete the last restore operation (in seconds)")
 )
 
+func init() {
+	for _, cmd := range []string{"mysqlctl", "mysqlctld", "vtadmin", "vtbackup", "vtcombo", "vtctl", "vtctld", "vttablet", "vttestserver"} {
```
Comment (ajm188) [inline go/vt/mysqlctl/backup.go:100]: i guarantee `vtadmin` does not need this flag. i know you did the `go list`, but that's just a preliminary list ... i think you need to trace through this one (which both I and @GuptaManan100 have had to do for others)

File: go/vt/mysqlctl/backupengine.go [hunk 2]
```diff
@@ -109,6 +112,16 @@ type BackupRestoreEngine interface {
 // BackupEngine and RestoreEngine.
 var BackupRestoreEngineMap = make(map[string]BackupRestoreEngine)
 
+func init() {
+	for _, cmd := range []string{"mysqlctl", "mysqlctld", "vtadmin", "vtbackup", "vtcombo", "vtctl", "vtctld", "vttablet", "vttestserver"} {
```
Comment (ajm188) [inline go/vt/mysqlctl/backupengine.go:116]: same comment r.e. tracing (and probably every other file in this package)

File: go/vt/mysqlctl/builtinbackupengine.go [hunk 3]
```diff
@@ -56,9 +59,16 @@ var (
 	// BuiltinBackupMysqldTimeout is how long ExecuteBackup should wait for response from mysqld.Shutdown.
 	// It can later be extended for other calls to mysqld during backup functions.
 	// Exported for testing.
-	BuiltinBackupMysqldTimeout = flag.Duration("builtinbackup_mysqld_timeout", 10*time.Minute, "how long to wait for mysqld to shutdown at the start of the backup")
+	//BuiltinBackupMysqldTimeout = flag.Duration("builtinbackup_mysqld_timeout", 10*time.Minute, "how long to wait for mysqld to shutdown at the start of the backup")
+
+	//builtinBackupProgress = flag.Duration("builtinbackup_progress", 5*time.Second, "how often to send progress updates when backing up large files")
```
Comment (ajm188) [inline go/vt/mysqlctl/builtinbackupengine.go:64]: why are these commented out?

File: go/vt/mysqlctl/mysqld.go [hunk 4]
```diff
@@ -53,30 +56,31 @@ import (
...
 var (
+
 	// DisableActiveReparents is a flag to disable active
 	// reparents for safety reasons. It is used in three places:
 	// 1. in this file to skip registering the commands.
 	// 2. in vtctld so it can be exported to the UI (different
 	// package, that's why it's exported). That way we can disable
 	// menu items there, using features.
-	DisableActiveReparents = flag.Bool("disable_active_reparents", false, "if set, do not allow active reparents. Use this to protect a cluster using external reparents.")
+	DisableActiveReparents = false
```
Comment (ajm188) [inline go/vt/mysqlctl/mysqld.go:66]: zero value is the default here

File: go/vt/mysqlctl/xtrabackupengine.go [hunk 5]
```diff
@@ -47,19 +50,19 @@ type XtrabackupEngine struct {
...
+	xbstreamRestoreFlags string
 	// streaming mode
-	xtrabackupStreamMode = flag.String("xtrabackup_stream_mode", "tar", "which mode to use if streaming, valid values are tar and xbstream")
-	xtrabackupUser       = flag.String("xtrabackup_user", "", "User that xtrabackup will use to connect to the database server. This user must have all necessary privileges. For details, please refer to xtrabackup documentation.")
+	xtrabackupStreamMode = "tar"
+	xtrabackupUser       string
 	// striping mode
-	xtrabackupStripes         = flag.Uint("xtrabackup_stripes", 0, "If greater than 0, use data striping across this many destination files to parallelize data transfer and decompression")
-	xtrabackupStripeBlockSize = flag.Uint("xtrabackup_stripe_block_size", 102400, "Size in bytes of each block that gets sent to a given stripe before rotating to the next stripe")
+	xtrabackupStripes         = uint(0)
```
Comment (ajm188) [inline go/vt/mysqlctl/xtrabackupengine.go:64]: zero value is the default

File: go/vt/mysqlctl/backupengine.go (hunk 2 above)
Comment (rsajwani) [reply go/vt/mysqlctl/backupengine.go:116]: I have tried my best to update this list .. besides I am assuming if I have forgot any binary then some of our test would high likely break, right ?

**Comment by ajm188** [comment]
i think you need to merge/rebase with main to get resolve the changes to `vtexplain.txt`, but otherwise LGTM whenever you're ready to move out of draft

File: go/vt/mysqlctl/backupengine.go [hunk 6]
```diff
@@ -19,14 +19,17 @@ package mysqlctl
...
 	"os"
 	"path"
 	"path/filepath"
 	"strings"
 	"time"
 
+	"github.com/spf13/pflag"
+
+	"vitess.io/vitess/go/vt/servenv"
+
```
Comment (ajm188) [inline go/vt/mysqlctl/backupengine.go:32]: nit: delete this blank

File: go/vt/mysqlctl/builtinbackupengine.go [hunk 7]
```diff
@@ -32,6 +31,10 @@ import (
 	"sync/atomic"
 	"time"
 
+	"github.com/spf13/pflag"
+
+	"vitess.io/vitess/go/vt/servenv"
+
```
Comment (ajm188) [inline go/vt/mysqlctl/builtinbackupengine.go:37]: nit: same

File: go/vt/mysqlctl/compression.go [hunk 8]
```diff
@@ -19,12 +19,15 @@ package mysqlctl
...
-	"flag"
 	"fmt"
 	"io"
 	"os/exec"
 	"sync"
 
+	"github.com/spf13/pflag"
+
+	"vitess.io/vitess/go/vt/servenv"
+
```
Comment (ajm188) [inline go/vt/mysqlctl/compression.go:30]: and here

File: go/vt/mysqlctl/mycnf_gen.go [hunk 9]
```diff
@@ -26,7 +26,9 @@ import (
 	"path"
 	"text/template"
 
-	"flag"
+	"github.com/spf13/pflag"
+
+	"vitess.io/vitess/go/vt/servenv"
```
Comment (ajm188) [inline go/vt/mysqlctl/mycnf_gen.go:32]: and here

File: go/vt/mysqlctl/mysqld.go [hunk 10]
```diff
@@ -41,6 +40,10 @@ import (
 	"sync"
 	"time"
 
+	"github.com/spf13/pflag"
+
+	"vitess.io/vitess/go/vt/servenv"
+
 	"vitess.io/vitess/config"
 
 	"vitess.io/vitess/go/mysql"
```
Comment (ajm188) [inline go/vt/mysqlctl/mysqld.go:49]: and here

File: go/vt/mysqlctl/mysqld.go [hunk 11]
```diff
@@ -53,30 +56,31 @@ import (
 )
 
 var (
+
```
Comment (ajm188) [inline go/vt/mysqlctl/mysqld.go:59]: and here

File: go/vt/mysqlctl/xtrabackupengine.go [hunk 12]
```diff
@@ -31,6 +30,10 @@ import (
 	"sync"
 	"time"
 
+	"github.com/spf13/pflag"
+
+	"vitess.io/vitess/go/vt/servenv"
+
```
Comment (ajm188) [inline go/vt/mysqlctl/xtrabackupengine.go:36]: and here

**Review by ajm188 — APPROVED** [review]
cleanup the whitespace changes, then LGTM!

**Pull request merged**